src/Foundatio/Utility/ScheduledTimer.cs

```diff
@@ -61,7 +63,8 @@ public class ScheduledTimer : IDisposable {
 
                 _logger.Trace(() => $"Scheduling next: delay={delay}");
 
-                _timer.Change(delay, Timeout.Infinite);
+                DisposeTimer();
+                _timer = SystemClock.Instance.Schedule(TimeSpan.FromMilliseconds(delay), async () => await RunCallbackAsync());
```

Comment:
I don't think it'll cause any race conditions that aren't already there. To be honest, though, I think this entire class could use a rewrite.